version: "3.7"
name: bytewax-redpanda
volumes:
  redpanda: null
services:
  redpanda:
    command:
      - redpanda start
      - --kafka-addr internal://0.0.0.0:9092
      - --advertise-kafka-addr internal://redpanda:9092
      - --pandaproxy-addr internal://0.0.0.0:8082
      - --advertise-pandaproxy-addr internal://redpanda:8082
      - --schema-registry-addr internal://0.0.0.0:8081
      - --rpc-addr redpanda:33145
      - --advertise-rpc-addr redpanda:33145
      - --smp 1
      - --memory 1G
      - --mode dev-container
      - --default-log-level=warn
    image: docker.redpanda.com/redpandadata/redpanda:v23.2.19
    container_name: redpanda
    volumes:
      - redpanda:/var/lib/redpanda/data
    ports:
      - "9092:9092"
    healthcheck:
      test: ["CMD-SHELL", "rpk cluster health | grep -E 'Healthy:.+true' || exit 1"]
      interval: 15s
      timeout: 3s
      retries: 5
      start_period: 5s
  dataflow:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: bytewax
    volumes:
      - ./main.py:/bytewax/main.py
    command: bash
    tty: true
    depends_on:
      redpanda:
        condition: service_healthy
